app/controllers/api/v1x0/graphql_controller.rb

```diff
@@ -17,7 +17,8 @@ def overlay
         {
           "^.*$" => {
             "base_query" => lambda do |model_class, _ctx|
-              "::Api::V1x0::#{model_class.to_s.pluralize}Controller".constantize.new.send(:index_scope, model_class.all)
+              authorize model_class
+              model_class
```

Comment:
@hsong-rh Should this be calling rbac_scope instead of authorize directly, Since you dont have the actual object to do the authorize on.
https://github.com/RedHatInsights/catalog-api/blob/0ae757293c7bfac2c1f02ddc49d8cc63e135245f/app/controllers/api/v1/graphql_controller.rb#L12

Reply:
@hsong-rh These are all index calls with read permission. The admin, approver and user could all make this call.

Reply:
`authorize ModelClass` is a class level permission check, not necessary on a certain instance. Here we just need the `read` permission check on `ModelClass` level

Reply:
@hsong-rh This is a sample GraphQL call note the ID inside

```
{
  requests(filter: {parent_id: "360"}) {
    id
    actions {
      id
      operation
      comments
      created_at
      processed_by
    }
    number_of_children
    decision
    description
    number_of_finished_children
    parent_id
  }
}
```
If you just check for the Class level access how would you prevent users from passing in any  parent_id and downloading the model.

Reply:
@mkanoor 👍 Will work on this kind of user case.